### PR TITLE
Add timezone-compatible date pickers

### PR DESCRIPTION
**Pull request by jordanwilking**
Add the timezone-compatible pickers and utilities that Andrew made as well as one with a MUI TextField. Replaces KeyBoardDatePickers.

The first commit has already been reviewed in #413 

Instructions for testing are downstream

### CODE REVIEW

File: ui-frontend/packages/catalog-ui-search/src/main/webapp/component/fields/date-time-picker.tsx [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+  React.useEffect(() => {
+    if (isInternalChange) {
+      setIsInternalChange(false)
+    } else {
+      // If the value is updated externally (i.e. a new value is passed into the component),
+      // then we need to update the display value.
+      // This allows us to keep the displayValue and value separate
+      // and type in the TextField without being hijacked
+      setDisplayValue(formatInputValue(value))
+    }
```
Comment (jordanwilking) [inline ui-frontend/packages/catalog-ui-search/src/main/webapp/component/fields/date-time-picker.tsx]: 📓 Before this, I had to deal with either:

1. Updating the calendar view wouldn't update the text field, or
1. The text field value would update as I typed

File: ui-frontend/packages/catalog-ui-search/src/main/webapp/component/fields/useTimePrefs.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { useState, useEffect } from 'react'
+import { useBackbone } from '../selection-checkbox/useBackbone.hook'
+// @ts-ignore ts-migrate(7016) FIXME: Could not find a declaration file for module '../s... Remove this comment to see the full error message
+import user from '../singletons/user-instance'
+
+const useTimePrefs = () => {
+  const { listenTo } = useBackbone()
+  const [, setForceRender] = useState(Math.random())
```
Comment (andrewkfiedler) [inline ui-frontend/packages/catalog-ui-search/src/main/webapp/component/fields/useTimePrefs.tsx:8]: Had no idea you could do this (leave out the first param).  🤯

File: ui-frontend/packages/catalog-ui-search/src/main/webapp/component/fields/date.tsx [hunk 3]
```diff
@@ -37,20 +34,10 @@ const validateShape = ({ value, onChange }: DateFieldProps) => {
 }
 
 export const DateField = ({ value, onChange, BPDateProps }: DateFieldProps) => {
-  const { listenTo } = useBackbone()
-  const [forceRender, setForceRender] = React.useState(Math.random())
+  useTimePrefs()
```
Comment (andrewkfiedler) [inline ui-frontend/packages/catalog-ui-search/src/main/webapp/component/fields/date.tsx:40]: Brilliant.

File: ui-frontend/packages/catalog-ui-search/src/main/webapp/component/fields/useTimePrefs.tsx (hunk 2 above)
Comment (vinamartin) [reply ui-frontend/packages/catalog-ui-search/src/main/webapp/component/fields/useTimePrefs.tsx:8]: ☝️ Same here lol, nice stuff

Comment (jordanwilking) [reply ui-frontend/packages/catalog-ui-search/src/main/webapp/component/fields/useTimePrefs.tsx:8]: I've had to learn to keep Typescript happy 😂

**Review by vinamartin — APPROVED** [review]
10/10 👏

File: ui-frontend/packages/catalog-ui-search/src/main/webapp/component/fields/useTimePrefs.tsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+  useEffect(() => {
+    listenTo(
+      user.getPreferences(),
+      'change:dateTimeFormat change:timeZone',
+      () => {
+        setForceRender(Math.random())
+      }
+    )
+  }, [])
```
Comment (stevenmalmgren) [inline ui-frontend/packages/catalog-ui-search/src/main/webapp/component/fields/useTimePrefs.tsx:18]: Do you need to stop listening to the model here?

**Review by stevenmalmgren — CHANGES_REQUESTED** [review]
See comment

File: ui-frontend/packages/catalog-ui-search/src/main/webapp/component/fields/useTimePrefs.tsx (hunk 4 above)
Comment (jordanwilking) [reply ui-frontend/packages/catalog-ui-search/src/main/webapp/component/fields/useTimePrefs.tsx:18]: @andrewkfiedler, I noticed that we don't stop listening in many places when using listenTo. Should we be doing that?

useBackbone does have a cleanup callback that stops listening, but I don't understand 100% what we're doing there. It looks like we create a new backbone model and alter its listenTo to have our backbone model as its `this1, but I'm not sure how that plays into the cleanup.

Comment (andrewkfiedler) [reply ui-frontend/packages/catalog-ui-search/src/main/webapp/component/fields/useTimePrefs.tsx:18]: We don't need to, the stopListening is available if necessary, but the useBackbone effect cleans up the most common usecase which is unlistening on destruction of the component.

**Comment by andrewkfiedler** [comment]
Hero Successful, tested both the advanced search input and the new input seen in the summary view (tried it on those without a date and those with a date).  User preferences are taken into account on both 🎉

**Pull request merged**